plugins/webhooks/index.js

```diff
@@ -634,53 +630,64 @@ async function createEvents(eventFactory, userFactory, pipelineFactory,
     const events = [];
     const meta = createMeta(parsed);
 
-    for (let i = 0; i < pipelines.length; i += 1) {
-        const p = pipelines[i];
-        /* eslint-disable no-await-in-loop */
-        const pipelineBranch = await p.branch;
-        /* eslint-enable no-await-in-loop */
-        const startFrom = determineStartFrom(action, type, branch, pipelineBranch);
+    const pipelineTaple = await Promise.all(pipelines.map(async (p) => {
+        const taple = { branch: await p.branch, pipeline: p };
+
+        return taple;
+    }));
+
+    const ignoreExtraTriggeredPipelines = pipelineTaple.filter((taple) => {
```

Comment:
`pipelineTable` is needed? Can we combine with L633-639 into one as below?
`const filteredPipelines = await Promise.all(pipelines.filter(async(p) => {...}))`

Reply:
Unfortunately, `filter` seems to not supported async. `filter` needs `boolean`, but it returns `Promise`.

Reply:
I got it, thanks!